### ISSUE TITLE
How to convert trained checkpoint back to zero123-xl.ckpt liked format？

### ISSUE THREAD

**Issue by AbrahamYabo** [kxhit/zero123-hf#4]
After training for args.checkpointing_steps times, two pytorch_model.bin, an optimizer.bin, a scheduler.bin and a random_states.pkl are saved at the log dir. How to convert these checkpoints back into the zero123-xl,ckpt liked format to produce 3D shapes? How to make the checkpoints that can be used by stable-dreamfusion or threestudio?

**Comment by kxhit** [comment]
What format exactly are you asking for? To convert back to diffusers format, just do `pipeline.save_pretrained(args.output)` should give you a folder that contains all submodels config.json and model.bin. So you can load to the pipeline and do inference again.

**Comment by AbrahamYabo** [comment]
I mean the original zero123 format like 105000.ckpt, 165000.ckpt and zero123-xl.ckpt. So that we can use https://github.com/ashawkey/stable-dreamfusion to create 3D model.

**Comment by AbrahamYabo** [comment]
Could you please provide a script (for example convert_diffusers_to_zero123.py) as convert_zero123_to_diffusers.py, that will help a lot. Thank you so much.

**Comment by kxhit** [comment]
Hi @AbrahamYabo 

Will try to have a look when I have time. It shouldn't be too hard to modify diffusers script [convert_diffusers_to_original_stable_diffusion.py](https://github.com/huggingface/diffusers/blob/main/scripts/convert_diffusers_to_original_stable_diffusion.py). You are welcome to contribute as well!

**Comment by AbrahamYabo** [comment]
Thanks for reply~

**Comment by xdobetter** [comment]
> Thanks for reply~

Hello, have you completed the conversion of the trained checkpoint to zero123-xl.ckpt format?

**Comment by AbrahamYabo** [comment]
> > Thanks for reply~
> 
> Hello, have you completed the conversion of the trained checkpoint to zero123-xl.ckpt format?

For Your Information:
https://github.com/AbrahamYabo/Cascade-Zero123/blob/main/scripts/convert_zero123_to_diffusers.py

**Comment by xdobetter** [comment]
Great work, Thank you very much for your effort!

**Comment by xdobetter** [comment]
> > > Thanks for reply~
> > 
> > 
> > Hello, have you completed the conversion of the trained checkpoint to zero123-xl.ckpt format?
> 
> For Your Information: https://github.com/AbrahamYabo/Cascade-Zero123/blob/main/scripts/convert_zero123_to_diffusers.py

Great work, Thank you very much for your effort!
